libs/ui/src/ui_strings/read_on_load.test.tsx

```diff
@@ -0,0 +1,140 @@
+import { Optional, assert } from '@votingworks/basics';
+import { act } from 'react-dom/test-utils';
+import React from 'react';
+import { createMemoryHistory } from 'history';
+import { Route, Router } from 'react-router-dom';
+import { ReadOnLoad } from './read_on_load';
+import { render, screen } from '../../test/react_testing_library';
+import {
+  UiStringsAudioContextInterface,
+  UiStringsAudioContextProvider,
+  useAudioContext,
+} from './audio_context';
+import {
+  UiStringsReactQueryApi,
+  createUiStringsApi,
+} from '../hooks/ui_strings_api';
+
+const mockUiStringsApi: UiStringsReactQueryApi = createUiStringsApi(() => ({
+  getAudioClips: jest.fn(),
+  getAvailableLanguages: jest.fn(),
+  getUiStringAudioIds: jest.fn(),
+  getUiStrings: jest.fn(),
+}));
+
+function renderWithClickListener(ui: React.ReactNode) {
+  const mockOnClick = jest.fn();
+  const result = render(<div onClickCapture={mockOnClick}>{ui}</div>);
+
+  return {
+    mockOnClick,
+    result,
+  };
+}
+
+let audioContext: Optional<UiStringsAudioContextInterface>;
+function TestContextConsumer() {
+  audioContext = useAudioContext();
+
+  return null;
+}
+
+afterEach(() => {
+  audioContext = undefined;
+});
+
+test('is no-op when audio context is absent', () => {
+  const { mockOnClick } = renderWithClickListener(
+    <ReadOnLoad>Bonjour!</ReadOnLoad>
```

Comment:
Completely unserious comment that doesn't actually require a code change, but I think that there's a space before exclamation marks in French 😝